Realm/RLMResults.mm

```diff
@@ -278,14 +287,14 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args {
 }
 
 - (RLMResults *)objectsWithPredicate:(NSPredicate *)predicate {
-    RLMResultsValidate(self);
+    RLMCheckThread(_realm);
 
     // copy array and apply new predicate creating a new query and view
     auto query = [self cloneQuery];
     RLMUpdateQueryWithPredicate(query.get(), predicate, _realm.schema, _realm.schema[self.objectClassName]);
     return [RLMResults resultsWithObjectClassName:self.objectClassName
                                             query:move(query)
-                                             sort:_backingView.m_sorting_predicate
+                                             sort:std::move(_sortOrder)
```

Comment:
Don't use `move` here. The Sorter needs to be copied into the new `RLMResults`, not moved from the current one into the new one.